src/main/java/de/tum/in/www1/artemis/service/ProgrammingExerciseGradingService.java

```diff
@@ -106,6 +110,60 @@ public ProgrammingExerciseGradingService(ProgrammingExerciseTestCaseService test
                 // This method will return without triggering the build if the submission is not of type TEST.
                 triggerTemplateBuildIfTestCasesChanged(programmingExercise.getId(), result.getId());
             }
+
+            if (!isSolutionParticipation && !isTemplateParticipation) {
+
+                Optional<ProgrammingExerciseStudentParticipation> studentParticipation = programmingExerciseParticipationService
+                        .findStudentParticipationWithLatestManualResultsAndFeedbacksAndRelatedSubmissionsAndAssessor(participation.getId());
+                if (studentParticipation.isPresent() && studentParticipation.get().getResults().stream().findFirst().isPresent()) {
+
+                    Result manualResult = studentParticipation.get().getResults().stream().findFirst().get();
+
+                    ProgrammingSubmission newSubmission = programmingSubmissionService.createSubmissionWithLastCommitHashForParticipation(participation, SubmissionType.MANUAL);
+                    Result newResult = programmingSubmissionService.setNewResult(newSubmission);
+                    newResult.setAssessor(manualResult.getAssessor());
+                    newResult.setAssessmentType(AssessmentType.SEMI_AUTOMATIC);
+                    newResult.setCompletionDate(manualResult.getCompletionDate());
+                    newResult.setHasFeedback(manualResult.getHasFeedback());
+                    newResult.setRated(manualResult.isRated());
+
+                    for (Feedback feedback : result.getFeedbacks()) {
```

Comment:
Maybe we can do it like we did it on `ProgrammingExerciseGradingService:462`

Reply:
Yes. 👍